pontoon/machinery/views.py

```diff
@@ -45,17 +45,18 @@ def machinery(request):
     })
 
 
-def translation_memory(request):
+def translation_memory(request=None, text=None, locale=None, pk=None):
     """Get translations from internal translations memory."""
-    try:
-        text = request.GET['text']
-        locale = request.GET['locale']
-        pk = request.GET['pk']
-    except MultiValueDictKeyError as e:
-        return JsonResponse({
-            'status': False,
-            'message': 'Bad Request: {error}'.format(error=e),
-        }, status=400)
+    if request:
+        try:
+            text = request.GET['text']
+            locale = request.GET['locale']
+            pk = request.GET['pk']
+        except MultiValueDictKeyError as e:
+            return JsonResponse({
+                'status': False,
+                'message': 'Bad Request: {error}'.format(error=e),
+            }, status=400)
```

Comment:
I don't like this too much. I'd keep the original function signature and then within it call `a_new_function(text, locale, pk)` that will also be used by `pre_translation`. @adngdb Do you agree?

Reply:
Yes, definitely. This is a django view, and should stay as such. If you want the insides to be shared, then extract the insides to a different function and call that function from the view.

Reply:
This way you also make the function call a lot less painful for the other places, because you don't have the JSON encoding logic and all.